shell.nix

```diff
@@ -0,0 +1,14 @@
+# Allows `nix-shell` without having to go trough the trouble of pinning the same
+# version as is done by the flake.
+{
+  pkgs ? import <nixpkgs> {},
+  checks ? {},
+}:
+pkgs.mkShell {
+  inputsFrom = builtins.attrValues checks;
+
+  buildInputs = with pkgs; [
+    cargo-flamegraph
+    rust-analyzer
+  ];
```

Comment:
Should things like `clippy` and `rustfmt` be mentioned here explicitly, or are they implied from the toolchain (which I presume is still a prerequisite)?

Reply:
A few lines above I include everything that we use in our tests. Including rustfmt and clippy. So they are indeed implied.